common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -419,7 +419,8 @@ protected void wakeup(boolean inEventLoop) {
 
     @Override
     public boolean inEventLoop(Thread thread) {
-        return thread == this.thread;
+        return thread == this.thread ||
+                executor instanceof EventExecutor && ((EventExecutor) executor).inEventLoop(thread);
```

Comment:
Yes this is more of a hack..I'm not generally pleased with this. Some deadlocks were silly sync calls I had forgotten about.  This problem was causing the execute method to spin because the thread was nulled out elsewhere and lazily set after this check failed. The intention was to use as much of the transport code as possible (Socket and common eventloops). I will look into the interface to swap out SingleThreadEventExecutor for an alternative.